src/write/mod.rs

```diff
@@ -292,22 +302,50 @@ impl Object {
 
     /// Append data to an existing section, and update a symbol to refer to it.
     ///
+    /// For Mach-O, this also creates a thread_var entry for TLS symbols.
```

Comment:
Please document that the symbol changes to the thread vars location when using Mach-O